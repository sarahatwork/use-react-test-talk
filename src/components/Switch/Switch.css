.text,
.image {
  display: block; 
  margin: 0 auto;
}