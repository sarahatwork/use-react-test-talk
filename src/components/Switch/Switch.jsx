import React from 'react';
import './Switch.css';

export default class Switch extends React.Component {
  state = {
    isOn: false
  };

  toggleSwitch = () => {
    this.setState({ isOn: !this.state.isOn });
  };

  render() {
    return (
      <div>
        <span data-testid="label" className="text">
          The toggle is: {this.state.isOn ? 'On' : 'Off'}
        </span>
        <button onClick={this.toggleSwitch}>Toggle</button>
      </div>
    );
  }
}
