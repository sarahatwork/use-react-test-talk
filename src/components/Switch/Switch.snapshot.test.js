import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Switch from './Switch';

describe('Switch test with Jest snapshot', () => {
  it('switches from off to on on user click', () => {
    const { asFragment, getByRole } = render(<Switch />);

    // save original output
    const firstRender = asFragment();

    // click button
    fireEvent.click(getByRole('button'));

    // compare to new output
    expect(firstRender).toMatchDiffSnapshot(asFragment());
  });
});
