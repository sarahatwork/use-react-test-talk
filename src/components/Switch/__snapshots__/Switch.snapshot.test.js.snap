// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch test with Jest snapshot is off by default 1`] = `
<section>
  <div>
    <span
      class="text"
      data-testid="label"
    >
      The toggle is: 
      Off
    </span>
    <button>
      Toggle
    </button>
  </div>
</section>
`;

exports[`Switch test with Jest snapshot switches from off to on on user click 1`] = `
<section>
  <div>
    <span
      class="text"
      data-testid="label"
    >
      The toggle is: 
      On
    </span>
    <button>
      Toggle
    </button>
  </div>
</section>
`;
